src/Adapter/MSTest.TestAdapter/Resources/Resource.resx

```diff
@@ -402,4 +402,9 @@ Error: {1}</value>
   <data name="STAIsOnlySupportedOnWindowsWarning" xml:space="preserve">
     <value>Runsettings entry '&lt;ExecutionApartmentState&gt;STA&lt;/ExecutionApartmentState&gt;' is not supported on non-Windows OSes.</value>
   </data>
+  <data name="TestMethodArgumentsMismatchError" xml:space="preserve">
+    <value>Argument mismatch error for test method '{0}.{1}': the number or types of arguments passed do not match the parameters.
+Arguments types: '{2}'
+Parameters types: '{3}'</value>
```

Comment:
@nohwnd Please feel free to suggest a different wording.

Reply:
```suggestion
    <value>Cannot run test: 
Parameters of test method '{0}.{1}' do not match the provided test data. Either the count or types are different.
Test expected {2} parameter(s), with types '{3}',
but received {4} argument(s), with types '{5}'.
```